src/interchange/src/avro.rs

```diff
@@ -896,16 +902,65 @@ fn build_schema(columns: &[(ColumnName, ColumnType)]) -> Schema {
             {
                 "name": "after",
                 "type": ["null", "row"]
+            },
+            {
+                "name": "transaction",
+                "type": [
+                    "null",
+                    {
+                        "name": "transaction_metadata",
+                        "type": "record",
+                        "fields": [
+                            {
+                                "name": "id",
+                                "type": "string",
+                            }
+                        ]
+                    }
+                ]
+            }
+        ]
```

Comment:
Mentioned elsewhere, but would it make sense to either omit the field or include it non-nullably based on whether the user has specified `consistency = true`?